app/upgrades/v15/upgrades.go

```diff
@@ -53,3 +60,43 @@ func migrateNextPoolId(ctx sdk.Context, gammKeeper *gammkeeper.Keeper, poolmanag
 		poolmanagerKeeper.SetPoolRoute(ctx, poolId, poolType)
 	}
 }
+
+func registerOsmoIonMetadata(ctx sdk.Context, bankKeeper bankkeeper.Keeper) {
+	uosmoMetadata := banktypes.Metadata{
+		Description: "The native token of Osmosis",
+		DenomUnits: []*banktypes.DenomUnit{
+			{
+				Denom:    appParams.BaseCoinUnit,
+				Exponent: 0,
+				Aliases:  nil,
+			},
+			{
+				Denom:    appParams.HumanCoinUnit,
+				Exponent: appParams.OsmoExponent,
+				Aliases:  nil,
+			},
+		},
+		Base:    appParams.BaseCoinUnit,
+		Display: appParams.HumanCoinUnit,
```

Comment:
Confirmed that these AppParams are const's, so no need to worry about anything in build editing them